### PR TITLE
backupccl: buffer/sort returned files before flushing to remote

### PR DESCRIPTION
**Pull request by dt**
BACKUP has multiple workers sending export requests concurrently so the
returned files may arrive out of order. Previously this would always
force the writer that flushed returned files out to the remote file to
close the file it had been writing an open a new one, as any one file
must be in-order. This adds a small queue of returned files to the sink
to which it adds files it is given to write to remote storage. Once the
queue has accumulated, it is sorted and partially drained to the remote
file. This should increase the odds that files are added to the remote
file in-order and thus do not require closing and re-opening additional
remote files.

With default worker (3) on a tpcc5k cluster running an incremental backup
this was observed to reduce the number of files a node wrote from ~50-70 to
<10.

Release note: none.

### CODE REVIEW

File: pkg/ccl/backupccl/backup_processor.go [hunk 1]
```diff
@@ -534,10 +562,53 @@ func (s *sstSink) Close() error {
 	return nil
 }
 
+// push pushes one returned backup file into the sink. Retutned files can arrive
```
Comment (pbardea) [inline pkg/ccl/backupccl/backup_processor.go:565]: s/Retutned/Returned

File: pkg/ccl/backupccl/backup_processor.go [hunk 2]
```diff
@@ -534,10 +562,53 @@ func (s *sstSink) Close() error {
...
+	}
+	return nil
+}
+
 func (s *sstSink) flush(ctx context.Context) error {
+	for i := range s.queue {
+		if err := s.write(ctx, s.queue[i]); err != nil {
+			return err
+		}
+	}
```
Comment (pbardea) [inline pkg/ccl/backupccl/backup_processor.go:602]: Does `s.queue` need to be reset here?

Comment (dt) [reply pkg/ccl/backupccl/backup_processor.go:603]: we don't reuse after flush, so I don't think so, but I guess it doesn't hurt.

**Review by pbardea — APPROVED** [review]

**Comment by dt** [comment]
TFTR!

bors r+

**Pull request merged**